packages/core/src/types.ts

```diff
@@ -1846,30 +1856,38 @@ export type __ResolvedTypesMetaFrom<T> = T extends StateMachine<
 export type EventOfMachine<TMachine extends AnyStateMachine> =
   TMachine extends StateMachine<any, infer E, any, any, any> ? E : never;
 
-export interface ActorContext<TEvent extends EventObject, TSnapshot> {
+export interface ActorContext<
+  TEvent extends EventObject,
+  TSnapshot,
+  TSystem extends ActorSystem<any> = ActorSystem<any>
```

Comment:
I wonder if this shouldn't accept the same generic as `ActorSystem` itself accepts. Not a big deal though and I don't have a strong opinion about this right now.

Reply:
Can change it later; if for whatever reason a dev needs to manually construct this, this type can be `typeof someSystem` rather than `ActorSystemInfoFrom<typeof someSystem>`